test/core/data.wast

```diff
@@ -333,3 +333,46 @@
 ;;   (module (memory 1) (data (global.get $g)) (global $g (mut i32) (i32.const 0)))
 ;;   "constant expression required"
 ;; )
+
+;; data segment with memory index 1 (only memory 0 available)
```

Comment:
For completeness I would also add a test with imported memory defined, and data segment with memory index 1.

Reply:
Added.